numba/cuda/dispatcher.py

```diff
@@ -662,22 +661,21 @@ def get_call_template(self, args, kws):
 
         A (template, pysig, args, kws) tuple is returned.
         """
-        with self._compiling_counter:
-            # Ensure an exactly-matching overload is available if we can
-            # compile. We proceed with the typing even if we can't compile
-            # because we may be able to force a cast on the caller side.
-            if self._can_compile:
-                self.compile_device(tuple(args))
+        # Ensure an exactly-matching overload is available if we can
```

Comment:
The differences seem to be:

- The base class also calls `pysig, args = self._compiler.fold_argument_types(args, kws)`
- This implementation needs to call `compile_device()` instead of `compile()`
- Because this implementation didn't call `fold_argument_types()`, it needs to use `utils.pysignature()` to get `pysig`.

I think it's a bit too different to consider uniting them at this stage.